src/printer.js

```diff
@@ -2062,8 +2062,12 @@ function printFunctionParams(path, print, options) {
     return concat(["(", join(", ", printed), ")"]);
   }
 
+  const isFlowShorthandWithOneArg = (isObjectTypePropertyAFunction(parent) ||
+    isTypeAnnotationAFunction(parent) || parent.type === "TypeAlias") &&
+    fun.params.length === 1 && fun.params[0].name === null && fun.rest === null;
```

Comment:
Looks like this has less cases than arrow function:

```js
        n.params.length === 1 &&
        !n.rest &&
        n.params[0].type === "Identifier" &&
        !n.params[0].typeAnnotation &&
        !n.params[0].leadingComments &&
        !n.params[0].trailingComments &&
        !n.params[0].optional &&
        !n.predicate &&
        !n.returnType
```

https://github.com/prettier/prettier/blob/master/src/printer.js#L304-L313

Can you add the following tests as well:

```js
type f = (...arg) => void;
type f = (/* comment */ arg) => void;
type f = (arg /* comment */) => void;
type f = (?arg) => void;
```